src/containerDeployment/dockerUtils.ts

```diff
@@ -0,0 +1,688 @@
+    GET_SQL_SERVER_CONTAINER_VERSIONS: `curl -s https://mcr.microsoft.com/v2/mssql/server/tags/list`,
+};
+
+/**
+ * The steps for the Docker container deployment process.
+ */
+export function initializeDockerSteps(): DockerStep[] {
+    return [
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: [],
+            headerText: ContainerDeployment.dockerInstallHeader,
+            bodyText: ContainerDeployment.dockerInstallBody,
+            errorLink: "https://docs.docker.com/engine/install/",
+            errorLinkText: ContainerDeployment.installDocker,
+            stepAction: checkDockerInstallation,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: [],
+            headerText: ContainerDeployment.startDockerHeader,
+            bodyText: ContainerDeployment.startDockerBody,
+            errorLink: "https://docs.docker.com/engine/",
+            errorLinkText: ContainerDeployment.startDockerEngine,
+            stepAction: startDocker,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: [],
+            headerText: ContainerDeployment.startDockerEngineHeader,
+            bodyText: ContainerDeployment.startDockerEngineBody,
+            stepAction: checkEngine,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: ["containerName", "password", "version", "hostname", "port"],
+            headerText: ContainerDeployment.creatingContainerHeader,
+            bodyText: ContainerDeployment.creatingContainerBody,
+            stepAction: startSqlServerDockerContainer,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: ["containerName"],
+            headerText: ContainerDeployment.settingUpContainerHeader,
+            bodyText: ContainerDeployment.settingUpContainerBody,
+            stepAction: checkIfContainerIsReadyForConnections,
+        },
+        {
+            loadState: ApiStatus.NotStarted,
+            argNames: [],
+            headerText: ContainerDeployment.connectingToContainerHeader,
+            bodyText: ContainerDeployment.connectingToContainerBody,
+            stepAction: undefined,
+        },
+    ];
+}
+
+/**
+ * Handles the result of a Docker command and updates the corresponding step statuses accordingly.
+ */
+export function setStepStatusesFromResult(
+    result: DockerCommandParams,
+    currentStep: DockerStepOrder,
+    steps: DockerStep[],
+): DockerStep[] {
+    if (result.success) {
+        steps[currentStep].loadState = ApiStatus.Loaded;
+    } else {
+        steps[currentStep].loadState = ApiStatus.Error;
+        steps[currentStep].errorMessage = result.error;
+        steps[currentStep].fullErrorText = truncateAndSanitizeErrorTextIfNeeded(
+            result.fullErrorText,
+        );
+        for (let i = currentStep + 1; i < steps.length; i++) {
+            steps[i].loadState = ApiStatus.Error;

```

Comment:
Why do we set the future steps to error. Shouldn't we just keep them NOT started?